src/run/editor/mod.rs

```diff
@@ -465,8 +465,7 @@ impl Editor {
 
     pub fn add_comparison<S: Into<String>>(&mut self, comparison: S) -> Result<(), ()> {
         let comparison = comparison.into();
-        if validate_comparison_name(&self.run, &comparison) {
-            self.run.add_custom_comparison(comparison);
+        if self.run.add_custom_comparison(comparison).is_ok() {
```

Comment:
You can use the ? operator to propagate the error up instead of using if ....().is_ok()